### ISSUE TITLE
UIA showing Expand/Collapsed property option on non-expandable components

### ISSUE THREAD

**Issue by YajurG** [microsoft/react-native-windows#11791]
### Problem Description

When focusing on non-expandable components (E.g Buttons, TouchableHighlight), the UIA tree shows the ExpandCollapsePattern as an option, when it should not be shown when focused on non expandable components.

Example:
![image](https://github.com/microsoft/react-native-windows/assets/30995198/f51eda0a-4b1f-4e21-8360-12cb18eeeb6d)

As can be seen, the ExpandCollapsePattern exists when focusing on a Button in RNTester.

### Steps To Reproduce

1. Run Playground and open RNTester
2. Navigate to the Button page in the Accessibility section of RNTester
3. Open Accessibility Insights and focus on any of the buttons.
4. Look at UIA Tree output.

### Expected Results

Ideally there should be no ExpandCollapsePattern in the UIA Tree for Buttons, TouchableHighlights and other non expandable or collapsible elements. 

### CLI version

12.0.0-alpha.3

### Environment

```markdown
System:
  OS: Windows 10 10.0.22621
  CPU: "(24) x64 AMD Ryzen Threadripper PRO 3945WX 12-Cores     "
  Memory: 45.90 GB / 63.86 GB
Binaries:
  Node:
    version: 18.12.1
    path: C:\Program Files\nodejs\node.EXE
  Yarn:
    version: 1.22.19
    path: C:\Program Files (x86)\Yarn\bin\yarn.CMD
  npm:
    version: 8.19.2
    path: C:\Program Files\nodejs\npm.CMD
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
  Windows SDK:
    AllowDevelopmentWithoutDevLicense: Enabled
    AllowAllTrustedApps: Enabled
    Versions:
      - 10.0.18362.0
      - 10.0.19041.0
      - 10.0.22000.0
      - 10.0.22621.0
IDEs:
  Android Studio: AI-213.7172.25.2113.9123335
  Visual Studio:
    - 17.6.33801.468 (Visual Studio Enterprise 2022)
    - 16.11.33529.622 (Visual Studio Enterprise 2019)
Languages:
  Java:
    version: 18.0.2.1
    path: C:\Program Files\Common Files\Oracle\Java\javapath\javac.EXE
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Target Platform Version

None

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2022

### Build Configuration

Debug
